sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthenticationRecord.java

```diff
@@ -108,13 +115,9 @@ public String getUsername() {
      * @return A {@link Mono} containing {@link Void}
      */
     public Mono<OutputStream> serializeAsync(OutputStream outputStream) {
-        return Mono.defer(() -> {
-            try {
-                OBJECT_MAPPER.writeValue(outputStream, this);
-            } catch (IOException e) {
-                return Mono.error(e);
-            }
-            return Mono.just(outputStream);
+        return Mono.fromCallable(() -> {
+            serialize(outputStream);
+            return outputStream; // this method's signature should have been Mono<Void>, so keep returning the outputStream.
```

Comment:
OutputStream is provided by user, we return so user can fluently continue using it and/or close it.